tests/file/test_handle.py

```diff
@@ -0,0 +1,153 @@
+            return
+
+        # Act
+        handle_generator = self.fs.list_handles(TEST_SHARE_NAME, recursive=True, max_results=1)
+
+        # Assert
+        self.assertIsNotNone(handle_generator.next_marker)
+        handles = list(handle_generator)
+        self._validate_handles(handles)
+
+        # Note down a handle that we saw
+        old_handle = handles[0]
+
+        # Continue listing
+        remaining_handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True, marker=handle_generator.next_marker))
+        self._validate_handles(handles)
+
+        # Make sure the old handle did not appear
+        # In other words, the marker worked
+        old_handle_not_present = all([old_handle.handle_id != handle.handle_id for handle in remaining_handles])
+        self.assertTrue(old_handle_not_present)
+
+    @record
+    def test_list_handles_on_directory(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, directory_name='wut', recursive=True))
+
+        # Assert
+        self._validate_handles(handles)
+
+    @record
+    def test_list_handles_on_file(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, directory_name='wut', file_name='bla.txt'))
+
+        # Assert
+        self._validate_handles(handles)
+
+    @record
+    def test_close_single_handle(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True))
+        self._validate_handles(handles)
+        handle_id = handles[0].handle_id
+
+        # Act
+        num_closed = list(self.fs.close_handles(TEST_SHARE_NAME, handle_id=handle_id))
+
+        # Assert 1 handle has been closed
+        self.assertEqual(1, num_closed[0])
+
+    @record
+    def test_close_all_handle(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True))
+        self._validate_handles(handles)
+
+        # Act
+        num_closed = list(self.fs.close_handles(TEST_SHARE_NAME, handle_id="*", recursive=True))
+
+        # Assert at least 1 handle has been closed
+        self.assertTrue(1 < num_closed[0])
```

Comment:
Any reason we can't assert the precise number of handles closed?

Reply:
(in each iteration)

Reply:
The test set up is highly manual, and the precise number depends on the share usage at that precise moment, so I didn't want to make the tests brittle.